### PR TITLE
Backport 18894: gui: Fix manual coin control with multiple wallets loaded

### PR DESCRIPTION
**Pull request by UdjinM6**
It's pretty hard to follow the logic in SendCoins and CoinControl dialogs, especially now that we have two send tabs. Turned out that there is also another issue - manual coin control isn't quite working with multiple wallets loaded but there is a patch in BTC which actually fixes both of them at once. Should not introduce any changes in the actual behaviour. Future merge conflicts are minimal and trivial as far as I can tell.

### CODE REVIEW

**Review by xdustinface — DISMISSED** [review]
ACK

File: src/qt/coincontroldialog.cpp [hunk 1]
```diff
@@ -41,10 +40,11 @@ bool CCoinControlWidgetItem::operator<(const QTreeWidgetItem &other) const {
     return QTreeWidgetItem::operator<(other);
 }
 
-CoinControlDialog::CoinControlDialog(QWidget* parent) :
+CoinControlDialog::CoinControlDialog(CCoinControl& coin_control, WalletModel* _model, QWidget *parent) :
```
Comment (PastaPastaPasta) [inline src/qt/coincontroldialog.cpp:43]: I lean towards not changing this, but this is a super nit
```suggestion
CoinControlDialog::CoinControlDialog(CCoinControl& coin_control, WalletModel* _model, QWidget* parent) :
```

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Review by xdustinface — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK
